sdcm/cluster.py

```diff
@@ -2881,8 +2884,12 @@ def __init__(self, cluster_uuid=None, cluster_prefix='cluster', node_prefix='nod
         self.log = SDCMAdapter(LOGGER, extra={'prefix': str(self)})
         self.log.info('Init nodes')
         self.nodes = []
-        self.instance_provision = params.get('instance_provision')
-        self.params = params
+        if params:
+            self.instance_provision = params.get('instance_provision')
+            self.params = params
+        else:
+            self.instance_provision = None
```

Comment:
when base cluster is not getting called with params ?

Reply:
In my local test i'am doing it all the times.
At the same time we define params=None, so we allow it to be None, but not covering this case in __init__

Reply:
then you have bug somewhere no passing down the params correctly

Reply:
I rather fix that, and assert params shouldn't be empty, you might be overwriting the `self.params` here, and breaking other code paths.

Reply:
It is needed for scylla_manager_cluster, which is PodCluster that have no params